### PR TITLE
OCPBUGS-3651: [release-4.12] cherry-pick hybrid overlay fixes from master branch 

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
cherry-pick: 82dcb005b966c1f71792a464bab038f54f991c8c, dabaf9fb8e024d8a52fbb1dc470fdce596caf505 the former did not apply cleanly but only needed a minor revision in AllocateHybridOverlay() the ladder did apply cleanly. 

This fixes allocation of the hybrid overlay DRIP by insuring that it will not collide with the IP of a new pod. 

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go [hunk 1]
```diff
@@ -292,8 +293,10 @@ func (manager *LogicalSwitchManager) AllocateHybridOverlay(nodeName string, hybr
 		for _, ip := range hybridOverlayAnnotation {
 			allocateAddresses = append(allocateAddresses, &net.IPNet{IP: net.ParseIP(ip).To4(), Mask: net.CIDRMask(32, 32)})
 		}
-		err := manager.AllocateIPs(nodeName, allocateAddresses)
-		if err != nil {
+		// attempt to allocate the IP address that is annotated on the node. The only way there would be a collision is if the annotations of podIP or hybridOverlayDRIP
+		// where manually edited and we do not support that
+		err := manager.AllocateIPs(switchName, allocateAddresses)
```
Comment (dcbw) [inline go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go:298]: @JacobTanenbaum 

```
 go test -mod=vendor -test.v -race  github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn -ginkgo.v  -ginkgo.reportFile /go/src/github.com/openshift/ovn-kubernetes/go-controller/_artifacts/junit-pkg_ovn.xml
# github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/logical_switch_manager
pkg/ovn/logical_switch_manager/logical_switch_manager.go:298:30: undefined: switchName
FAIL	github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn [build failed] 
```

switchName => nodeName?

Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/logical_switch_manager/logical_switch_manager.go]: done

**Review by trozet — APPROVED** [review]
/lgtm

**Pull request merged**